### DISCUSSION TITLE
Size mismatch error when using a model trained in a recognition model

### DISCUSSION THREAD

**Discussion by yerin723** [mindee/doctr#1330] [General]
hi? I am now trying to learn recognition by creating several png file images and the corresponding labels.json file.
I'm trying to run a test with an existing pt file before learning, but what is the error below?

```python
Traceback (most recent call last):
  File "/home/piai/Desktop/ocr/doctr/references/recognition/train_pytorch.py", line 482, in <module>
    main(args)
  File "/home/piai/Desktop/ocr/doctr/references/recognition/train_pytorch.py", line 242, in main
    model.load_state_dict(checkpoint)
  File "/home/piai/Desktop/ocr/venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 2041, in load_state_dict
    raise RuntimeError('Error(s) in loading state_dict for {}:\n\t{}'.format(
RuntimeError: Error(s) in loading state_dict for CRNN:
	size mismatch for linear.weight: copying a param with shape torch.Size([124, 256]) from checkpoint, the shape in current model is torch.Size([101, 256]).
	size mismatch for linear.bias: copying a param with shape torch.Size([124]) from checkpoint, the shape in current model is torch.Size([101]).
``` 

I ran the code below.

!python3 /home/piai/Desktop/ocr/doctr/references/recognition/train_pytorch.py --train_path /home/piai/Desktop/ocr/dataset/recognition/train --val_path /home/piai/Desktop/ocr/dataset/recognition/test "crnn_vgg16_bn" --epochs 20 -b 4 --device 0 --lr 0.005 -j 8 --pretrained --resume "/home/piai/Desktop/ocr/pretrained_weight/crnn_vgg16_bn-9762b0b0.pt" --test-only

**Comment by felixT2K** [comment]
Hi @yerin723 👋🏼 ,

looks like you use another vocab as the default (french) you can specify it with `--vocab=german` for example.
You can find the full list of available voabs at https://mindee.github.io/doctr/modules/datasets.html#supported-vocabs
Keep in mind each character in your labels should exist in the used vocab.
Btw.: `--pretrained --resume "/home/piai/Desktop/ocr/pretrained_weight/crnn_vgg16_bn-9762b0b0.pt" ` does the same it's enough to use `--pretrained`
I have added the ability to freeze the backbone also for recognition models in https://github.com/mindee/doctr/pull/1328 this makes it easier to fine tune (only european -> non european needs training from scratch)

**Reply by yerin723** [reply]
My label has the following format:

    "forged_roll_1_2.png": "243",
    "forged_roll_1_3.png": "449",
    "forged_roll_1_4.png": "8",
    "forged_roll_1_5.png": "2",
    "forged_roll_1_6.png": "42",
    "forged_roll_1_8.png": "85",
    "forged_roll_1_9.png": "2",
    "forged_roll_1_11.png": "50",
    "forged_roll_1_12.png": "120",
    "forged_roll_1_13.png": "0.5",
    "forged_roll_1_h_38.png": "pi81450",

How do I set up vocab?
I should recogntion the numbers well.
    
I'm trying to train a dataset containing integers, real numbers, and the symbol pi.
VOCABS["test"] = VOCABS["digits"] + VOCABS["punctuation"] + VOCABS["symbol"]

Why does an error occur even if I input it like this?
